src/control/server/storage/provider.go

```diff
@@ -185,11 +185,15 @@ func (p *Provider) ControlMetadataIsMounted() (bool, error) {
 		return false, errors.New("nil provider")
 	}
 
-	p.log.Debugf("control metadata config: %+v", p.engineStorage.ControlMetadata)
 	if !p.engineStorage.ControlMetadata.HasPath() {
 		// If there's no control metadata path, we use SCM for control metadata
-		return p.ScmIsMounted()
+		if isMounted, err := p.ScmIsMounted(); os.IsNotExist(errors.Cause(err)) {
+			return false, nil // Mountpoint maybe missing
```

Comment:
Some logging message could be useful here

Reply:
TBH I don't think we need extra logging here, the aim of the change is to prevent a missing mountpoint from being confused as a missing superblock. This case will indicate that the mountpoint is not mounted and further down the chain the path existence will be checked explicitly.

Reply:
Instead of returning nil here, IMO we should return a more informative, helpful error. Masking the error with the assumption that it gets caught later doesn't seem like the right approach to me.

Reply:
My thinking behind the change is that `ControlMetadataIsMounted()` shouldn't really be worried about whether the mountpoint exists or not, either it is mounted or not. If the mountpoint doesn't exist then it's simply not mounted so that should be the return result. Returning NotExist in the context of the IsMounted check doesn't make sense to me. I don't think this is masking the error, I see it as making the function output more intuitive.